packages/parser/src/language/sankey/sankeyMatcher.ts

```diff
@@ -0,0 +1,91 @@
+import { accessibilityDescrRegex, accessibilityTitleRegex, titleRegex } from '../common/matcher.js';
+
+import type {
+  CustomPatternMatcherFunc,
+  CustomPatternMatcherReturn,
+  IToken,
+} from '../chevrotainWrapper.js';
+
+/**
+ * Matches sankey link source and target node
+ */
+export const sankeyLinkNodeRegex = /(?:"((?:""|[^"])+)")|([^\n\r,]+(?=%%)|[^\n\r,]+)/;
+
+/**
+ * Matches sankey link value
+ */
+export const sankeyLinkValueRegex = /("(?:0|[1-9]\d*)(?:\.\d+)?"|[\t ]*(?:0|[1-9]\d*)(?:\.\d+)?)/;
+
+type CustomMatchResult = CustomPatternMatcherReturn | RegExpExecArray | null;
+
+const STICKY_FLAG = 'y';
+
+/**
+ * Try to match the text with a SankeyNodeLink.
+ *
+ * These have to be checked first because they take precedence.
+ * Langium does not provide any way to specify precedence for grammars (or parts thereof)
+ * that are `imported` into another grammar definition file (e.g. sankey.langium).
+ * Specifically, the order of _tokens_ defined in the imported file (e.g. common.langium)
+ * may or may not come before (or after) the _tokens_ defined in the including file (e.g. sankey.langium).
+ *
+ * Thus, we have to manually handle this by first checking for matches that should take precedence
+ * (in this case title, accessibility title, and accessibility description)
+ * over matching this token.
+ *
+ * First check if the text matches a title, accessibility title, or accessibility description.
+ * If it matches one of those, return null (no match with a SankeyNodeLink).
+ *
+ * If it does not match one of those, then check to see if the text matches the
+ * SankeyNodeLink Regexp.
+ *
+ * Note that _all_ regular expressions have the sticky flag set.
+ *
+ * @param text - text to check for matches
+ * @param startingOffset - offset to start at
+ * @param _tokens - ITokens previously matched (unused)
+ * @param _groups - named grammar groups (unused)
+ *
+ * @returns  Null if there is no match, else return the RegExpExecArray with the match.
+ */
+export const matchSankeyLinkNode: CustomPatternMatcherFunc = (
+  text: string,
+  startingOffset = 0,
+  _tokens: IToken[],
+  _groups
+): CustomMatchResult => {
+  const regexpsToFail: RegExp[] = [accessibilityDescrRegex, accessibilityTitleRegex, titleRegex];
```

Comment:
I'm still pretty unfamiliar with Langium, I'm sure you're all more experienced than me, but this approach looks good to me.

Although we may want to consider defining this variable in `packages/parser/src/language/common/matcher.ts`, especially since then we can use something like https://github.com/spadgos/regex-combiner to create one optimized regex instead of three separate ones.

In the long term, it might be worth asking Langium to add some options about importing tokens and the precedence. I can imagine that other people might encounter the same issues as us, and having syntax like `import "../common/common" with xxxx;` might make sense.